typescript-definitions-derive/src/tests.rs

```diff
@@ -3,108 +3,285 @@ use super::Typescriptify;
 
 #[cfg(test)]
 mod macro_test {
-    // use proc_macro2::Ident;
     use super::Typescriptify;
-    use insta::assert_snapshot_matches;
     use quote::quote;
+
+    macro_rules! assert_conversion {
+        ($tokens:expr,$expected:expr) => {{
+            let declarations = Typescriptify::new($tokens)
+                .parse()
+                .export_type_definition_source()
+                .declarations;
+
+            assert_eq!(declarations, $expected)
+        }};
+    }
+
+    // The crate only converts complex types, so we need to wrap the type in a struct to check its
+    // conversion. We could also write these tests at the conversion logic inside the crate but
+    // right now I'm just tring to get as much value for as little effort as possible.
+    macro_rules! assert_type_conversion {
+        ($rust:ty,$ts:literal) => {
```

Comment:
I think I got the write macro designators but not 100% sure.

Reply:
Can you help me understand this a little more? In all cases I think we're wanting to test a Type conversion. Are you thinking we could reuse this for other tests, or am I misunderstanding?

For context, my very limited experience comes from: https://medium.com/@phoomparin/a-beginners-guide-to-rust-macros-5c75594498f1

Reply:
Ah, I think what you're saying is that the `type` designator will match anything that has a valid `type` syntax, rather than being an actual type?

Reply:
The difference is between something that *identifies* a type versus something that *creates* it. The former is matched as `ty`, the latter is always an `item` (because all type declarations are items).

Reply:
We discussed this and there was a bit of misunderstanding about the intent. We're happy with `ty`.